test/java/com/arpnetworking/metrics/portal/alerts/impl/FileAlertRepositoryTest.java

```diff
@@ -0,0 +1,160 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arpnetworking.metrics.portal.alerts.impl;
+
+import com.arpnetworking.metrics.portal.TestBeanFactory;
+import com.arpnetworking.testing.SerializationTestUtils;
+import com.arpnetworking.utility.test.ResourceHelper;
+import com.google.common.collect.ImmutableMap;
+import models.internal.Organization;
+import models.internal.alerts.Alert;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.collection.IsMapWithSize.anEmptyMap;
+import static org.junit.Assert.fail;
+
+/**
+ * Unit Tests for {@link FileAlertRepository}.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+public class FileAlertRepositoryTest {
+    private static final UUID METADATA_ALERT_ID = UUID.fromString("0eca730a-5f9a-49db-8711-29a49cac98ff");
+    private static final Map<String, Object> EXPECTED_METADATA = ImmutableMap.of(
+            "externalFieldA", "A",
+            "externalFieldB", ImmutableMap.of(
+                    "externalFieldC", "C"
+            )
+    );
+
+    private Organization _organization;
+    private FileAlertRepository _repository;
+
+    @Before
+    public void setUp() throws IOException, URISyntaxException {
+        final URL url = ResourceHelper.resourceURL(FileAlertRepositoryTest.class, "Alerts");
+        _organization = TestBeanFactory.createOrganization();
+        _repository = new FileAlertRepository(
+                SerializationTestUtils.getApiObjectMapper(),
+                Paths.get(url.toURI()),
+                _organization.getId()
+        );
+        _repository.open();
+    }
+
+    @After
+    public void tearDown() {
+        _repository.close();
+    }
+
+    @Test
+    public void testInvalidPathOnOpen() {
+        final Path path;
+        try {
+            path = Paths.get("valid_but_nonexistent");
+        } catch (final InvalidPathException e) {
+            fail("Test path should be valid");
+            return; // Needed to convince the compiler path is initialized.
+        }
```

Comment:
I think you could avoid this dance by adding `... throws Exception` to the signature. (I would never suggest this for _production_ code, but it seems A-okay to me in the context of this test, since (a) nobody ever _calls_ it, and (b) throwing an exception will cause it to fail, which is what we want.)